File: src/csharp/api/Models/Prescription.cs
using Example.Api.Enums;

namespace Example.Api.Models;

/// <summary>
/// Represents a prescription in the system.
/// </summary>
/// <value></value>
public record Prescription : BaseEntity
{
    /// <summary>
    /// The unique identifier for the prescription.
    /// </summary>
    /// <value></value>
    public long Id { get; set; }

    /// <summary>
    /// The OrderId associated with the prescription.
    /// </summary>
    /// <value></value>
    public long OrderId { get; set; }

    /// <summary>
    /// The MedicationId prescribed in the prescription.
    /// </summary>
    /// <value></value>
    public long MedicationId { get; set; }

    /// <summary>
    /// The dosage information for the prescription.
    /// </summary>
    /// <value></value>
    public string? Dose { get; set; }

    /// <summary>
    /// The frequency of administration for the prescription.
    /// </summary>
    /// <value></value>
    public string? Frequency { get; set; }

    /// <summary>
    /// The duration in days for the prescription.
    /// </summary>
    /// <value></value>
    public int DurationInDays { get; set; }

    /// <summary>
    /// The route of administration for the prescription.
    /// </summary>
    /// <value></value>
    public MedicationRoute Route { get; set; }

    /// <summary>
    /// The medication associated with the prescription.
    /// </summary>
    /// <value></value>
    public Medication? Medication { get; set; }
}

File: src/python/app/entities/prescription.py
"""Prescription entity model."""

from __future__ import annotations
from typing import Optional
from app.db.types import IntEnumType
from app.entities.base import Base
from app.entities.base_entity import BaseEntity
from app.entities.medication import Medication
from app.enums.medication_route import MedicationRoute
from sqlalchemy import ForeignKey, Integer, String, Index, BigInteger
from sqlalchemy.orm import Mapped, mapped_column, relationship


class Prescription(Base, BaseEntity):
    """Prescription model."""

    __tablename__ = "prescription"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    order_id: Mapped[int] = mapped_column(
        BigInteger, ForeignKey("patient_order.id", ondelete="RESTRICT"), nullable=False
    )
    medication_id: Mapped[int] = mapped_column(
        BigInteger, ForeignKey("medication.id", ondelete="RESTRICT"), nullable=False
    )

    dose: Mapped[Optional[str]] = mapped_column(String(50), nullable=True)
    frequency: Mapped[Optional[str]] = mapped_column(String(50), nullable=True)
    duration_in_days: Mapped[int] = mapped_column(Integer, nullable=False)

    route: Mapped[MedicationRoute] = mapped_column(IntEnumType(MedicationRoute), nullable=False)

    # Relationships
    # Prescription has medication navigation (many -> one). C# Prescription does not expose an Order navigation.
    medication: Mapped["Medication"] = relationship("Medication", back_populates="prescriptions")


# Indexes to match EF Core configuration
Index("ix_prescription_order_id", "order_id")
Index("ix_prescription_medication_id", "medication_id")
